mypy/main.py

```diff
@@ -110,10 +110,10 @@ def main(
         print_memory_profile()
 
     code = 0
-    if messages:
+    n_errors, n_notes, n_files = util.count_stats(messages)
+    if messages and n_notes < len(messages):
```

Comment:
nit: from the impl, it looks like it's not super guaranteed that `n_errors + n_notes == len(messages)`, maybe clearer to phrase as `if messages and n_errors`?